sdk/src/main/java/com/qonversion/android/sdk/internal/QProductCenterManager.kt

```diff
@@ -349,9 +364,42 @@ internal class QProductCenterManager internal constructor(
         }
 
         purchasingCallbacks[purchaseModel.product.storeID] = callback
+
+        updatePurchaseOptions(purchaseModel.options, purchaseModel.product.storeID)
+
         billingService.purchase(context, purchaseModel)
     }
 
+    @VisibleForTesting
+    fun actualPurchaseOptions(): Map<String, QPurchaseOptions> {
+        processingPurchaseOptions?.let {
+            return it
+        }
+
+        val cachedPurchaseOptions = purchasesCache.loadProcessingPurchasesOptions()
+        processingPurchaseOptions = cachedPurchaseOptions
+
+        return cachedPurchaseOptions
+    }
```

Comment:
I offer to make it as follows:

```
private val processingPurchaseOptions: MutableMap<String, QPurchaseOptions>? by lazy {
    purchasesCache.loadProcessingPurchasesOptions()?.toMutableMap()
}
```

and use the mutable map, without recreating it every time.

Reply:
I don't want to use a mutable map here to avoid modifying it from anywhere.

Reply:
You can modify it in the only method as you do now